indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```
@@ -0,0 +1,2770 @@
+                                          latestDataSourceMetadata.getSeekableStreamPartitions() != null &&
+                                          ioConfig.getId().equals(
+                                              latestDataSourceMetadata.getSeekableStreamPartitions().getStream()
+                                          );
+    final Map<PartitionType, SequenceType> latestOffsetsFromDb;
+    if (hasValidOffsetsFromDb) {
+      latestOffsetsFromDb = latestDataSourceMetadata.getSeekableStreamPartitions().getPartitionSequenceNumberMap();
+    } else {
+      latestOffsetsFromDb = null;
+    }
+
+    // order tasks of this taskGroup by the latest sequenceId
+    taskSequences.sort((o1, o2) -> o2.rhs.firstKey().compareTo(o1.rhs.firstKey()));
+
+    final Set<String> tasksToKill = new HashSet<>();
+    final AtomicInteger earliestConsistentSequenceId = new AtomicInteger(-1);
+    int taskIndex = 0;
+
+    while (taskIndex < taskSequences.size()) {
+      TreeMap<Integer, Map<PartitionType, SequenceType>> taskCheckpoints = taskSequences.get(taskIndex).rhs;
+      String taskId = taskSequences.get(taskIndex).lhs;
+      if (earliestConsistentSequenceId.get() == -1) {
+        // find the first replica task with earliest sequenceId consistent with datasource metadata in the metadata
+        // store
+        if (taskCheckpoints.entrySet().stream().anyMatch(
+            sequenceCheckpoint -> sequenceCheckpoint.getValue().entrySet().stream().allMatch(
+                partitionOffset -> {
+                  OrderedSequenceNumber<SequenceType> offset = makeSequenceNumber(partitionOffset.getValue());
+                  OrderedSequenceNumber<SequenceType> latestOffset = makeSequenceNumber(
+                      latestOffsetsFromDb == null ? partitionOffset.getValue() :
+                      latestOffsetsFromDb.getOrDefault(
+                          partitionOffset
+                              .getKey(),
+                          partitionOffs
```

Definitely don't swallow the exception here